### ISSUE TITLE
model-graded evaluation with python script

### ISSUE THREAD

**Issue by Rusteam** [promptfoo/promptfoo#610]
We're using an internal proxy to forward requests to chatgpt. I want to use for grading the outputs. I've tried to configure it in the following way:
```
defaultTest:
  options:
    provider: 
      - id: python:scripts/compass.py
        config:
          model: gpt-4
```
with calling the internal api inside python.
However, I face this issue:
![image](https://github.com/promptfoo/promptfoo/assets/22130831/fec11cc6-4998-4d37-bd67-574c8985dab4)

**Comment by typpo** [comment]
It looks like you are creating an array when you don't mean to.  It should look like this:

```yaml
defaultTest:
  options:
    provider: 
      id: python:scripts/compass.py
      config:
        model: gpt-4
```

**Comment by Rusteam** [comment]
@typpo  thanks, i've tried the way you suggested, however got another error:
```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at Object.join (node:path:1175:7)
    at PythonProvider.callApi (/opt/homebrew/lib/node_modules/promptfoo/dist/src/providers/pythonCompletion.js:22:63)
    at matchesLlmRubric (/opt/homebrew/lib/node_modules/promptfoo/dist/src/matchers.js:227:38)
    at async runAssertion (/opt/homebrew/lib/node_modules/promptfoo/dist/src/assertions.js:663:17)
    at async /opt/homebrew/lib/node_modules/promptfoo/dist/src/assertions.js:82:24
```

**Comment by Rusteam** [comment]
When I use it in a providers list (not for grading) it works fine though

**Comment by typpo** [comment]
Hmm.. can you try this and let me know how it works?
```yaml
defaultTest:
  options:
    provider: 
      id: python:scripts/compass.py
      config:
        model: gpt-4
        basePath: ''
```

**Comment by Rusteam** [comment]
unfortunately, same error.

I see you've committed a fix, have you tested it?

**Comment by typpo** [comment]
Fix is released in [0.51.0](https://github.com/promptfoo/promptfoo/releases/tag/0.51.0)!
